File: javascript/7kyu/Replace every nth.js
// https://www.codewars.com/kata/57fcaed83206fb15fd00027a

// Write a method, that replaces every nth char oldValue with char newValue.

function replaceNth(text, n, oldValue, newValue) {
    const array = text.split("");
    let counter = 1;
    
    for(let i = 0; i < array.length; i++){
        if(array[i] == oldValue && counter == n){
            array[i] = newValue;
            counter = 1;
        }
        if(array[i] == oldValue && counter != n){
            counter++;
        }
    }
    
    return array.join("");
}

File: java/7kyu/Replace every nth.java
// https://www.codewars.com/kata/57fcaed83206fb15fd00027a

// Write a method, that replaces every nth char oldValue with char newValue.

public class Kata {
  public static String replaceNth(String text, int n, char oldValue, char newValue) {
    int copy = n;
    char[] c = text.toCharArray();
    for (int i = 0; i < c.length; i++) {
      if (copy == 1 && c[i] == oldValue) {
        c[i] = newValue;
        copy = n;
      }
      else if(c[i] == oldValue) copy--;
    }
    return new String(c);
  }
}